packages/angular_devkit/build_angular/src/builders/browser-esbuild/index.ts

```diff
@@ -74,36 +75,58 @@ function normalizeOptions(options: BrowserBuilderOptions): ApplicationBuilderOpt
 // and not output browser files into '/browser'.
 async function writeResultFiles(
   outputFiles: BuildOutputFile[],
-  assetFiles: { source: string; destination: string }[] | undefined,
+  assetFiles: BuildOutputAsset[] | undefined,
   outputPath: string,
 ) {
   const directoryExists = new Set<string>();
-  await Promise.all(
-    outputFiles.map(async (file) => {
-      // Ensure output subdirectories exist
-      const basePath = path.dirname(file.path);
-      if (basePath && !directoryExists.has(basePath)) {
-        await fs.mkdir(path.join(outputPath, basePath), { recursive: true });
-        directoryExists.add(basePath);
-      }
-      // Write file contents
-      await fs.writeFile(path.join(outputPath, file.path), file.contents);
-    }),
-  );
+  const ensureDirectoryExists = async (basePath: string) => {
+    if (basePath && !directoryExists.has(basePath)) {
+      await fs.mkdir(path.join(outputPath, basePath), { recursive: true });
+      directoryExists.add(basePath);
+    }
+  };
+
+  // Writes the output file to disk and ensures the containing directories are present
+  await emitFilesToDisk(outputFiles, async (file: BuildOutputFile) => {
+    const fullOutputPath = file.fullOutputPath;
+    // Ensure output subdirectories exist
+    const basePath = path.dirname(fullOutputPath);
+    await ensureDirectoryExists(basePath);
+
+    // Write file contents
+    await fs.writeFile(path.join(outputPath, fullOutputPath), file.contents);
+  });
 
   if (assetFiles?.length) {
-    await Promise.all(
-      assetFiles.map(async ({ source, destination }) => {
-        // Ensure output subdirectories exist
-        const basePath = path.dirname(destination);
-        if (basePath && !directoryExists.has(basePath)) {
-          await fs.mkdir(path.join(outputPath, basePath), { recursive: true });
-          directoryExists.add(basePath);
-        }
-        // Copy file contents
-        await fs.copyFile(source, path.join(outputPath), fsConstants.COPYFILE_FICLONE);
-      }),
-    );
+    await emitFilesToDisk(assetFiles, async (asset: BuildOutputAsset) => {
+      // Ensure output subdirectories exist
+      await ensureDirectoryExists(asset.destination);
+
+      // Copy file contents
+      await fs.copyFile(
+        asset.source,
+        path.join(outputPath, asset.destination),
+        fsConstants.COPYFILE_FICLONE,
+      );
+    });
+  }
+}
+
+const MAX_CONCURRENT_WRITES = 64;
```

Comment:
Maybe give option to increase this value with `--max-concurrent-writes=100`?

Reply:
Adding an option is not something that we plan to do. Instead we opted for a sensible default.